### PR TITLE
Leave benefit selections blank on pdf if unselected

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Summary

If a claimant leaves  Burial and Funeral Costs, Cost of the Plot, or Transportation Costs unselected in the online form, they should not result in a checked box on the PDF. 

Previously, if the transportation benefit option was unselected, it appeared as "no" on the pdf. The VBA is concerned that a claimant may not have meant to opt out of the benefit, so leaving it unchecked on the pdf will leave the door open for additional benefits in the future.

 - If Burial and Funeral Costs is left blank, field 22A should appear blank.
- If Cost of the plot (gravesite) or interment is left blank, field 23 should appear blank.
- If Transportation costs are left blank, field 24 will appear blank.

Field 22A is set by the question "Are you responsible for the Veteran’s burial expenses?" (which has yes and no radio options), and field 23 is set by the followup question "Are you responsible for the Veteran’s plot or interment expenses?" (which has yes and no radio options). The question "Which of these costs did you pay for?" only determines whether to show the followup questions for these two fields. 

As such, the only field which required changes was the Transportation Costs option. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/80639

## Testing done

- [x] *New code is covered by unit tests*
- [x] Submitted the form with transportation unchecked, and confirmed that the generated PDF did not check the 'no' option

## Screenshots

Screenshot 2024-04-12 at 10 21 44 AM

## What areas of the site does it impact?
Burials V2

## Acceptance criteria

- [x]  Benefit selections that are left unchecked on the online form should remain unchecked on the pdf
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by tblackwe — APPROVED** [review]
Good change, TYVM

File: spec/fixtures/pdf_fill/21P-530V2/merge_fields.json [hunk 1]
```diff
@@ -93,7 +94,5 @@
     "hasPreviouslyReceivedAllowance":"YES",
     "noPreviouslyReceivedAllowance":null,
     "hasAllowanceStatementOfTruth":null,
-    "noAllowanceStatementOfTruth":null,
-    "hasTransportation":"YES",
-    "noTransportation":null
```
Comment (rmtolmach) [inline spec/fixtures/pdf_fill/21P-530V2/merge_fields.json:98]: Do these need to be removed from `lib/pdf_fill/forms/va21p530v2.rb`?

Comment (TaiWilkin) [reply spec/fixtures/pdf_fill/21P-530V2/merge_fields.json:98]: Only 'noTransportation' was removed; 'hasTransportation' was just moved up, and 'noAllowanceStatementOfTruth' had a comma removed. 'noTransportation' has been removed from `lib/pdf_fill/forms/va21p530v2.rb`

Comment (rmtolmach) [reply spec/fixtures/pdf_fill/21P-530V2/merge_fields.json:98]: oh! I see now. Thanks 😄

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**